packages/eslint-plugin/docs/rules/no-extraneous-class.md

```diff
@@ -2,14 +2,21 @@
 
 Forbids the use of classes as namespaces.
 
-This rule warns when a class is accidentally used as a namespace.
+This rule warns when a class has no non-static members, such as for a class used exclusively as a static namespace.
 
 ## Rule Details
 
-From TSLint’s docs:
+Users who come from a [OOP](https://en.wikipedia.org/wiki/Object-oriented_programming) paradigm may wrap their utility functions in an extra class, instead of putting them at the top level of an ECMAScript module.
+Doing so is generally unnecessary in JavaScript and TypeScript projects.
 
-> Users who come from a Java-style OO language may wrap their utility functions in an extra class,
-> instead of putting them at the top level.
+- Wrapper classes add extra runtime bloat and cognitive complexity to code without adding any structural improvements
```

Comment:
@goktugyil https://github.com/typescript-eslint/typescript-eslint/issues/5040#issuecomment-1134013764:

> Also how does this add cognitive complexity?

The standard approach of exporting items from an ES module has just one level of organization to keep them within: the ES Module itself. Adding an additional class they have to be a field on increases that count from 1 to 2.

Additionally, you have the slightly more complex rules around when static fields may be created or modified to take into account. [Class static initialization blocks](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Class_static_initialization_blocks) are sometimes necessary to initialize values. 

Does all that seem reasonable to you?